tests/mutation_test.cpp

```diff
@@ -412,3 +412,39 @@ TEST_CASE( "The various type of triggers work", "[mutations]" )
     }
 
 }
+
+TEST_CASE( "All valid mutations can be purified", "[mutations][purifier]" )
+{
+    std::vector<mutation_branch> dummies;
+    std::vector<mutation_branch> valid_traits;
```

Comment:
While it shouldn't have any behavioural difference, you shouldn't be copying `mutation_branches` like this (it's both wrong and inefficient), you should use `trait_id`s.
```suggestion
    std::vector<trait_id> dummies;
    std::vector<trait_id> valid_traits;
```